Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1087 @@
+	{
+		
+		//AddButtonEventHandler( button, UIE_CHANGE, OnSettingButtonPressed  )
+		// get panel
+		var panel = file.modPanels[i]
+
+		// reset to default nav
+		var child = Hud_GetChild( panel, "BtnMod" )
+
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "BtnMod" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "BtnMod" ) )
+
+		// Enum button nav
+		child = Hud_GetChild( panel, "EnumSelectButton" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "main" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "next" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "prev" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "EnumSelectButton" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "EnumSelectButton" ) )
+		Hud_AddEventHandler( child, UIE_CLICK, UpdateEnumSetting )
+
+		// reset button nav
+		
+		child = Hud_GetChild( panel, "ResetModToDefault" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "ResetModToDefault" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "ResetModToDefault" ) )
+
+		Hud_AddEventHandler( child, UIE_CLICK, ResetConVar )
+		
+		// text field nav
+		child = Hud_GetChild( panel, "TextEntrySetting" )
+
+		// 
+		Hud_AddEventHandler( child, UIE_LOSE_FOCUS, SendTextPanelChanges )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "TextEntrySetting" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "TextEntrySetting" ) )
+
+		child = Hud_GetChild( panel, "Slider" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "Slider" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "Slider" ) )
+
+		file.sliders.append(MS_Slider_Setup(child))
+
+		Hud_AddEventHandler( child, UIE_CHANGE, OnSliderChange )
+
+		child = Hud_GetChild( panel, "OpenCustomMenu" )
+
+		Hud_AddEventHandler( child, UIE_CLICK, CustomButtonPressed )
+	}
+
+	//Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnModsSearch" ), UIE_LOSE_FOCUS, OnFilterTextPanelChanged )
+	Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnFiltersClear" ), UIE_CLICK, OnClearButtonPressed )
+	// mouse delta 
+	AddMouseMovementCaptureHandler( file.menu, UpdateMouseDeltaBuffer )
+
+	Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnModsSearch" ), UIE_CHANGE, void function ( var inputField ) : (){
+		file.filterText = Hud_GetUTF8Text( inputField )
+		OnFiltersChange(0)
+	})
+}
+
+// magic function that converts an index from the full list into a index of a button,
+// BUT loops around when getting an out-of-view index (so, when showing 0-7, 8 will return 0 and -1 will return 7)
+// used for navigation code.
+int function GetIndex( int index, int length )
+{
+	if (index < 0)
+		return (length - 1) - (-index - 1) % length // this is really weird
+	return index % length
```

Comment:
There are a couple of ways that this could be improved imo, you seem to have developed a solution that works for both negative and positive numbers, and then only used it for negative numbers

You could either simplify the equations
```suggestion
	if (index < 0)
		return index % length + length
	return index % length
```
Or you could remove the if statement as it seems unnecessary
```suggestion
	return (length - 1) - (-index - 1) % length // this is really weird
```

I think both of these solutions work, but without doing any testing beyond plugging numbers in and having a look I can't say for 100% sure

Reply:
It is used for both positive and negative numbers, see [menu_mod_settings#157-191](https://github.com/R2Northstar/NorthstarMods/pull/518/files#diff-d8f519140adbffbd6fe54140bc2105cc69bd73066ad8bd1ff584ba3ca3f1dc9bR157-R191)

Reply:
> It is used for both positive and negative numbers, see [menu_mod_settings#157-191](https://github.com/R2Northstar/NorthstarMods/pull/518/files#diff-d8f519140adbffbd6fe54140bc2105cc69bd73066ad8bd1ff584ba3ca3f1dc9bR157-R191)

Sorry, i should specify that i mean that that line 219 is only used for negative numbers, due to the `if (index < 0)` directly above it

Reply:
Either way, function has been replaced so this is now irrelevant really